post.md

```diff
@@ -0,0 +1,226 @@
+${pin}= Get Request gpio_server /api/v1/pin/18 ${headers}
+Log  ${pin.json()}
+Delete All Sessions
+```
+
+For getting all signals you can only type hiperlink without pin number.
+Example:
+
+```
+Get signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin ${headers}
+Log ${pinout.json()}
+```
+
+That was easier part of our script.
+
+To set high or low state for GPIO's you should use `patch` method.
+First step is creating dictionary with value to send. It is not difficult:
+```
+${message}= Create Dictionary value 0
+```
+
+`Patch` request requires message data for send.
+
+```
+Set 0 signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${message}= Create Dictionary value 0
+${pin}= Patch Request gpio_server /api/v1/pin/18 ${message} headers=${headers}
+Log ${pin.json()}
+Delete All Sessions
+```
+
+We can use the same test to set the high state on GPIO. For indicator I used LED
+diodes connected serial with resistors.
+
+Full version of my code
+
+```
+*** Settings ***
+Library RequestsLibrary
+Library Collections
+
+
+*** Variables ***
+${URL} http://192.168.0.100:5000
+
+*** Test Cases ***
+
+Get signal
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin/18 ${headers}
+Log ${pinout.json()}
+
+Set value 1
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${message}= Create Dictionary value 1
+${pinout}= Patch Request gpio_server /api/v1/pin/18 ${message} headers=${headers}
+Log ${pinout.json()}
+Sleep 3 seconds
+
+Set value 0
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${message}= Create Dictionary value 0
+${pinout}= Patch Request gpio_server /api/v1/pin/18 ${message} headers=${headers}
+Log ${pinout.json()}
+
+Get all signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin ${headers}
+Log ${pinout.json()}
+```
+
+This code is not in perfect form, but it's not the subject of this article.
+So polishing is left to the reader.
+RF generates very readable reports after all tests are done.
+
+Reports look like this:
+![](http://3mdeb.com/wp-content/uploads/2017/10/report.png)
+
+You can see that Robot Framework is very easy to use environment for testing REST
+API. All method (e.g. POST and Patch) are the same in use. If you need
+help with implementing tests you can ask me or chat with users on slack 
+-`requests` channel. I help new users, when I can, and also sometimes I'm 
```

Comment:
and sometimes

Reply:
`whenever I can`  and  `I'm the one looking for help` sound much better.